TLM/TLM/Util/ModsCompatibilityChecker.cs

```diff
@@ -1,82 +1,201 @@
-using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Linq;
-using System.Reflection;
-using System.Text;
 using ColossalFramework;
-using ColossalFramework.PlatformServices;
+using ColossalFramework.PlatformServices; // used in RELEASE builds
 using ColossalFramework.Plugins;
 using ColossalFramework.UI;
 using CSUtil.Commons;
+using ICities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq; // used in RELEASE builds
+using System.Reflection;
 using TrafficManager.UI;
-using UnityEngine;
+using static ColossalFramework.Plugins.PluginManager;
+
+namespace TrafficManager.Util
+{
+    public class ModsCompatibilityChecker
+    {
 
-namespace TrafficManager.Util {
-    public class ModsCompatibilityChecker {
-        //TODO include %APPDATA% mods folder
+        public const ulong LOCAL_MOD = ulong.MaxValue;
 
+        // Used for LoadIncompatibleModsList()
         private const string RESOURCES_PREFIX = "TrafficManager.Resources.";
-        private const string DEFAULT_INCOMPATIBLE_MODS_FILENAME = "incompatible_mods.txt";
-        private readonly ulong[] userModList;
-        private readonly Dictionary<ulong, string> incompatibleModList;
+        private const string INCOMPATIBLE_MODS_FILE = "incompatible_mods.txt";
 
-        public ModsCompatibilityChecker() {
-            incompatibleModList = LoadIncompatibleModList();
-            userModList = GetUserModsList();
+        // parsed contents of incompatible_mods.txt
+        private readonly Dictionary<ulong, string> incompatibleMods;
+
+        public ModsCompatibilityChecker()
+        {
+            incompatibleMods = LoadListOfIncompatibleMods();
         }
 
-        public void PerformModCheck() {
-            Log.Info("Performing incompatible mods check");
-            Dictionary<ulong, string> incompatibleMods = new Dictionary<ulong, string>();
-            for (int i = 0; i < userModList.Length; i++) {
-                string incompatibleModName;
-                if (incompatibleModList.TryGetValue(userModList[i], out incompatibleModName)) {
-                    incompatibleMods.Add(userModList[i], incompatibleModName);
-                }
+        /// <summary>
+        /// Initiates scan for incompatible mods. If any found, and the user has enabled the mod checker, it creates and initialises the modal dialog panel.
+        /// </summary>
+        public void PerformModCheck()
```

Comment:
What is the caller of this method? The `PluginManager` may have an incomplete list of mods in particular game states.

Reply:
https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/pull/333/files#diff-ab18d276b8fa6e073177650dfcf8cc8cR58

Reply:
Added exception catching/logging so if any errors occur it will just silently abort the mod checker.